protocol/synthetix/contracts/storage/DelegationIntent.sol

```diff
@@ -0,0 +1,166 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.11 <0.9.0;
+
+import "./Config.sol";
+import "./Pool.sol";
+
+import "../interfaces/IVaultModule.sol";
+
+/**
+ * @title Represents a delegation (or undelegation) intent.
+ */
+library DelegationIntent {
+    using Pool for Pool.Data;
+
+    bytes32 private constant _ATOMIC_VALUE_LATEST_ID = "delegateIntent_idAsNonce";
+
+    /**
+     * Intent Lifecycle:
+     *
+     *          |<---- Delay ---->|<-- Processing Window -->|
+     * Time ----|-----------------|-------------------------|---->
+     *          ^                 ^                         ^
+     *          |                 |                         |
+     *  declarationTime    processingStartTime    processingEndTime
+     *
+     * Key:
+     * - declarationTime: Timestamp at which the intent is declared.
+     * - processingStartTime: Timestamp from which the intent can start being processed.
+     * - processingEndTime: Timestamp after which the intent cannot be processed.
+     *
+     * The intent can be processed only between processingStartTime and processingEndTime.
+     */
+    struct Data {
+        /**
+         * @notice An incrementing id (nonce) to ensure the  uniqueness of the intent and prevent replay attacks
+         */
+        uint256 id;
```

Comment:
this field could be done without--see other comments

delegation is already very expensive so every storage slot we can reduce is huge help